lib/phoenix/live_dashboard/components/nav_bar_component.ex

```diff
@@ -36,7 +47,53 @@ defmodule Phoenix.LiveDashboard.NavBarComponent do
         raise ArgumentError, msg <> inspect(no_list)
 
       {:ok, items} ->
-        %{items: normalize_items(items)}
+        nav_param = normalize_nav_param(params)
+
+        %{
+          items: normalize_items(items),
+          nav_param: nav_param,
+          extra_params: normalize_extra_params(params, nav_param)
+        }
+    end
+  end
+
+  defp normalize_extra_params(params, nav_param) do
+    case Map.fetch(params, :extra_params) do
+      :error ->
+        []
+
+      {:ok, extra_params_list} when is_list(extra_params_list) ->
```

Comment:
This makes me wonder: nav_param is an atom but extra_params is a list of strings. Should they be consistent?

Reply:
I changed to be always atoms, but internally I replace the `extra_params` to binaries in order to easily compare with `params`.